pkg/ebpf/tracer_windows.go

```diff
@@ -2,23 +2,149 @@
+	"time"
+	"unsafe"
+
 	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	DRIVERFILE = `\\.\ddfilter`
+
+	// https://github.com/DataDog/datadog-windows-filter/blob/master/include/ddfilterapi.h
+	DDFILTER_IOCTL_GETSTATS               = 0x801
+	DDFILTER_IOCTL_SIMULATE_COMPLETE_READ = 0x802
+	DDFILTER_IOCTL_SET_FILTER             = 0x803
+
+	// https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/specifying-device-types
+	NETWORK_DEVICE_TYPE_CTL_CODE = 0x00000012
+)
+
+var (
+	kernel32        = syscall.MustLoadDLL("kernel32.dll")
+	CreateFile      = kernel32.MustFindProc("CreateFileW")
+	CloseHandle     = kernel32.MustFindProc("CloseHandle")
+	expvarEndpoints map[string]*expvar.Map
+	expvarTypes     = []string{"driver_total_stats", "driver_handle_stats"}
 )
 
+func init() {
+	expvarEndpoints = make(map[string]*expvar.Map, len(expvarTypes))
+	for _, name := range expvarTypes {
+		expvarEndpoints[name] = expvar.NewMap(name)
+	}
+}
+
 // Tracer struct for tracking network state and connections
 type Tracer struct {
-	config *Config
+	config       *Config
+	driverHandle syscall.Handle
 }
 
 // NewTracer returns an initialized tracer struct
 func NewTracer(config *Config) (*Tracer, error) {
-	return &Tracer{}, nil
+	handle, err := openDriverFile(DRIVERFILE)
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "Could not create driver handle", err)
+	}
+	tr := &Tracer{
+		driverHandle: handle,
+	}
+	go tr.expvarStats()
+	return tr, nil
+}
+
+func (t *Tracer) expvarStats() {
+	ticker := time.NewTicker(5 * time.Second)
+	// starts running the body immediately instead waiting for the first tick
+	for ; true; <-ticker.C {
+		stats, err := t.GetStats()
+		if err != nil {
+			continue
+		}
+
+		for name, stat := range stats {
+			for metric, val := range stat.(map[string]int64) {
+				currVal := &expvar.Int{}
+				currVal.Set(val)
+				expvarEndpoints[name].Set(snakeToCapInitialCamel(metric), currVal)
+			}
+		}
+	}
 }
 
 // Stop function stops running tracer
 func (t *Tracer) Stop() {}
 
+func openDriverFile(path string) (syscall.Handle, error) {
+	p, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return syscall.InvalidHandle, err
+	}
+	log.Debug("Creating Driver File...")
+	r, _, err := CreateFile.Call(uintptr(unsafe.Pointer(p)),
+		syscall.GENERIC_READ,
+		syscall.FILE_SHARE_READ|syscall.FILE_SHARE_WRITE,
+		0,
+		syscall.OPEN_EXISTING,
+		syscall.FILE_FLAG_OVERLAPPED,
+		0)
+	log.Debug("Creating Driver Handle...")
+	h := syscall.Handle(r)
+	if h == syscall.InvalidHandle {
+		return h, err
+	}
+	log.Info("Connected to driver and handle created")
+	return h, nil
+}
+
+func closeDriverFile(handle syscall.Handle) error {
+	r, _, err := CloseHandle.Call(uintptr(handle))
+	if r == 0 {
+		return err
+	}
+	return nil
+}
+
+func getIoCompletionPort(handleFile syscall.Handle) (syscall.Handle, error) {
+	iocpHandle, err := syscall.CreateIoCompletionPort(handleFile, 0, 0, 0)
```

Comment:
same

Reply:
Updated - resolving. If i misinterpreted what you were saying, let me know.